src/main/java/com/autotune/database/dao/ExperimentDAOImpl.java

```diff
@@ -75,6 +78,57 @@ public ValidationOutputData addExperimentToDB(KruizeExperimentEntry kruizeExperi
         return validationOutputData;
     }
 
+    /**
+     * Deletes database partitions based on a specified threshold day count.
+     * <p>
+     * This method iterates through all Kruize tables in the database, extracts the date part
+     * from their names, and compares it with a cutoff date = currentDate - thresholdDaysCount. Tables with dates before the cutoff
+     * date are eligible for deletion, and the method executes SQL statements to drop these tables.
+     *
+     * @param thresholdDaysCount The number of days to be used as the threshold for partition deletion.
+     *                           Tables with dates older than this threshold will be deleted.
+     * @throws RuntimeException if any exception occurs during the deletion process. The exception
+     *                          details are logged, and the deletion process continues for other tables.
+     */
+    @Override
+    public void deletePartitions(int thresholdDaysCount) {
+        LOGGER.info("Threshold is set to {}", thresholdDaysCount);
+        // Calculate the date 'daysCount' days ago
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, -thresholdDaysCount);
+        Date cutoffDate = calendar.getTime();
+        String yyyyMMdd = "yyyyMMdd";
+        try (Session session = KruizeHibernateUtil.getSessionFactory().openSession()) {
+            List<String> tablenames = session.createNativeQuery(SELECT_ALL_KRUIZE_TABLES).getResultList();
+            for (String tableName : tablenames) {   // Since tableName cannot be null, there is no need to implement null handling; it can be skipped.
+                String datePart = null;
+                if (tableName.startsWith(DBConstants.TABLE_NAMES.KRUIZE_RESULTS + "_")) {
+                    datePart = tableName.substring((DBConstants.TABLE_NAMES.KRUIZE_RESULTS + "_").length());
+                } else if (tableName.startsWith(DBConstants.TABLE_NAMES.KRUIZE_RECOMMENDATIONS + "_")) {
+                    datePart = tableName.substring((DBConstants.TABLE_NAMES.KRUIZE_RECOMMENDATIONS + "_").length());
+                }
+                if (null != datePart) {
+                    Date tableDate = new SimpleDateFormat(yyyyMMdd).parse(datePart);
+                    // Compare the date part with the cutoffDate  (cutoffDate = todaysDate - thresholdDaysCount)
+                    if (tableDate.after(cutoffDate)) {
```

Comment:
What happens to the boundary condition, when the date is the cutoffDate, are we supposed to skip that?

Reply:
if today date is Jan15th 2024 and thresholdDaysCount = 15
then cutOffDate = Dec 31st 2023

and tables belongs to cutOffDate will get deleted , Answer is  it will not skip

Reply:
to avoid border situation we have option to set thresholdDaysCount using env varaible